src/tracker-component.js

```diff
@@ -186,16 +186,16 @@ export const Tracker = (config? : Config = {}) => {
 
     try {
         getOrCreateValidCartId();
+        getOrCreateValidUserId();
+
         if (config && config.user && config.user.id) {
-            setUserId(config.user.id);
-        } else {
-            getOrCreateValidUserId();
+            setMerchantProvidedUserId(config.user.id);
         }
     } catch (err) {
         // eslint-disable-next-line no-console
         console.error(err.message);
         createNewCartId();
-        createNewUserId();
+        setGeneratedUserId();
```

Comment:
do we need both ` setGeneratedUserId()` and `getOrCreateValidUserId`?

Reply:
`getOrCreateValidUserId` is a convenience method that's useful in Tracker initialization